test/functional/test_framework/mempool_util.py

```diff
@@ -39,19 +42,20 @@ def fill_mempool(test_framework, node, miniwallet):
     # Generate UTXOs to flood the mempool
     # 1 to create a tx initially that will be evicted from the mempool later
     # 75 transactions each with a fee rate higher than the previous one
-    test_framework.generate(miniwallet, 1 + (num_of_batches * tx_batch_size))
+    ephemeral_miniwallet = MiniWallet(node, tag_name="fill_mempool_ephemeral_wallet")
+    test_framework.generate(ephemeral_miniwallet, 1 + num_of_batches * tx_batch_size)
```

Comment:
For my understanding: I get that having an ephemeral wallet created and destroyed inside this function gets rid of any data sharing related to `MiniWallet` between the calls of `fill_mempool()`. Since the same tag name is being passed and subsequently the same `internal_key` is generated, how does adding the tag name really help here?

Reply:
I wouldn't say it disambiguates between separate `fill_mempool` calls, just between `fill_mempool` and the other `MiniWallet`s. I suppose we can add a parameter to modify the tag name further, but we don't have any functional tests where we do this multiple times.

Reply:
> where we do this multiple times

Reply:
`fill_mempool`

Reply:
Oh I see.